src/StatusQ/Controls/StatusBaseInput.qml

```diff
@@ -54,91 +56,107 @@ Item {
-
-        Flickable {
-            id: flick
+        border.color: {
+            if (!statusBaseInput.valid) {
+                return Theme.palette.dangerColor1;
+            }
+            if (edit.activeFocus) {
+                return Theme.palette.primaryColor1;
+            }
+            return sensor.containsMouse ? Theme.palette.primaryColor2 : "transparent"
+        }
 
+        MouseArea {
+            id: sensor
+            hoverEnabled: true
             anchors.fill: parent
-            anchors.leftMargin: statusBaseInput.leftPadding
-            anchors.rightMargin: statusBaseInput.rightPadding + clearable ? clearButtton.width : 0
-            anchors.topMargin: statusBaseInput.topPadding
-            anchors.bottomMargin: statusBaseInput.bottomPadding
-            contentWidth: edit.paintedWidth
-            contentHeight: edit.paintedHeight
-            clip: true
-
-            QC.ScrollBar.vertical: QC.ScrollBar { interactive: multiline }
-
-
-            function ensureVisible(r) {
-                if (contentX >= r.x)
-                    contentX = r.x;
-                else if (contentX+width <= r.x+r.width)
-                    contentX = r.x+r.width-width;
-                if (contentY >= r.y)
-                    contentY = r.y;
-                else if (contentY+height <= r.y+r.height)
-                    contentY = r.y+r.height-height;
-            }
+            cursorShape: Qt.IBeamCursor
+            onClicked: edit.forceActiveFocus()
+            Flickable {
+                id: flick
+
+                anchors.fill: parent
+                anchors.leftMargin: statusBaseInput.leftPadding
+                anchors.rightMargin: statusBaseInput.rightPadding + clearable ? clearButtton.width : 0
+                anchors.topMargin: statusBaseInput.topPadding
+                anchors.bottomMargin: statusBaseInput.bottomPadding
+                contentWidth: edit.paintedWidth
+                contentHeight: edit.paintedHeight
+                clip: true
+
+                QC.ScrollBar.vertical: QC.ScrollBar { interactive: multiline }
+
+
+                function ensureVisible(r) {
+                    if (contentX >= r.x)
+                        contentX = r.x;
+                    else if (contentX+width <= r.x+r.width)
+                        contentX = r.x+r.width-width;
+                    if (contentY >= r.y)
+                        contentY = r.y;
+                    else if (contentY+height <= r.y+r.height)
+                        contentY = r.y+r.height-height;
+                }
 
 
 
-            TextEdit {
-                id: edit
-                width: flick.width
-                selectByMouse: true
-                anchors.verticalCenter: parent.verticalCenter
-                focus: true
+                TextEdit {
+                    id: edit
+                    width: flick.width
+                    selectByMouse: true
```

Comment:
I just tested your code and found that I cannot right click and do Cut/Copy/Paste (context menu).
Is this something you will be introducing later?

Reply:
Yea something that I haven't even thought about to be honest. Wasn't area that built-in/native right click menus don't work here. cc'ing @simonricoo to potentially update the design system.